### PR TITLE
Fixes new tf32 failures in test_nn.py

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Also modify the `tf32_on_and_off` decorator to make it support function without `device` argument.

### CODE REVIEW

File: torch/testing/_internal/common_nn.py [hunk 1]
```diff
@@ -3692,7 +3693,7 @@ def fractional_max_pool3d_test(test_case):
         check_gradgrad=False,
         desc='gelu_activation',
         with_tf32=True,
-        tf32_precision=0.01,
+        tf32_precision=0.05,
```
Comment (ngimel) [inline torch/testing/_internal/common_nn.py:3696]: Isn't it worrying? Did the new cublas version come out?

Comment (zasdfgbnm) [reply torch/testing/_internal/common_nn.py:3696]: This is the cublas in CUDA 11.2.1 which we have never looked through the tests greenish.

File: torch/testing/_internal/common_cuda.py [hunk 2]
```diff
@@ -109,25 +109,23 @@ def with_tf32_enabled(self, function_call):
...
-                else:
-                    f(self, device, dtype)
+        params = inspect.signature(f).parameters
+        arg_names = tuple(params.keys())
+
+        @functools.wraps(f)
+        def wrapped(*args, **kwargs):
+            for k, v in zip(arg_names, args):
+                kwargs[k] = v
+            cond = tf32_is_not_fp32()
```
Comment (mruberry) [inline torch/testing/_internal/common_cuda.py:119]: Add a comment explaining the computation of cond and its effect

Comment (mruberry) [reply torch/testing/_internal/common_cuda.py:119]: Also update the comment explaining the decorator so readers know when it has an effect

Comment (zasdfgbnm) [reply torch/testing/_internal/common_cuda.py:133]: I just added all comments in front of this function.

**Comment by zasdfgbnm** [comment]
ping @ngimel @mruberry

**Comment by mruberry** [comment]
@zasdfgbnm Thank you for the ping. This looks good to me now. @ngimel, would you like to take a look?

**Review by mruberry — APPROVED** [review]
Thanks @zasdfgbnm!

**Pull request closed without merging**